src/app/component/shared/components/calendar-base/calendar-base.component.ts

```diff
@@ -213,39 +237,167 @@ export class CalendarBaseComponent implements OnInit, OnDestroy {
+      return `${dayItem.date.getFullYear()}-${dayItem.date.getMonth() + 1 < 10 ?
         '0' + (dayItem.date.getMonth() + 1) : dayItem.date.getMonth() + 1}-${dayItem.date.getDate() < 10 ?
-        '0' + dayItem.date.getDate() : dayItem.date.getDate()}`;
+          '0' + dayItem.date.getDate() : dayItem.date.getDate()}`;
+    }
+  }
+
+  formatSelectedDate(isMonthCalendar, dayItem: CalendarInterface) {
+    if (isMonthCalendar) {
+      return this.months[dayItem.month] + ' ' + dayItem.numberOfDate + ', ' + dayItem.year;
+    } else {
+      const month = dayItem.date.toLocaleDateString(this.language, { month: 'long' });
+      const day = dayItem.date.getDate();
+      const year = dayItem.date.getFullYear();
+      return month + ' ' + day + ', ' + year;
+    }
+  }
+
+  getHabitsForDay(habitsList, date) {
+    const habitsListForDay = habitsList.find(list => list.enrollDate === date);
+    return habitsListForDay;
+  }
+
+  public getUserHabits(isMonthCalendar, days) {
+    let firstDay;
+    if (isMonthCalendar) {
+      firstDay = days.find(day => day.numberOfDate === 1);
+    } else {
+      firstDay = days[0];
+    }
+    const startDate = this.formatDate(isMonthCalendar, firstDay);
+    const endDate = this.formatDate(isMonthCalendar, days[days.length - 1]);
+    this.habitAssignService.getAssignHabitsByPeriod(startDate, endDate).pipe(
+      takeUntil(this.destroyGetHabits)
+    ).subscribe(res => {
+      this.userHabitsList = res;
+      days.forEach(day => {
+        const date = this.formatDate(isMonthCalendar, day);
+        if (new Date().setHours(0, 0, 0, 0) >= new Date(date).setHours(0, 0, 0, 0)) {
+          day.hasHabitsInProgress = this.userHabitsList.filter(h => h.enrollDate === date)[0].habitAssigns.length > 0;
+          day.areHabitsDone = this.userHabitsList
+            .filter(h => h.enrollDate === date)[0].habitAssigns
+            .filter(hab => !hab.enrolled).length === 0;
+        }
+      });
+    });
+  }
+
+  isCheckedAllHabits(habitsForDay) {
+    const unenrolledHabit = habitsForDay.find(h => h.enrolled === false);
+    if (unenrolledHabit) {
+      return false;
+    } else {
+      return true;
+    }
+  }
+
+  chooseDisplayClass(dayItem) {
+    if (dayItem.isCurrentDayActive) {
+      this.getClass = 'current-day';
+      return this.getClass;
+    } else if (dayItem.hasHabitsInProgress && dayItem.numberOfDate < (dayItem.date.getDate() - 7) && dayItem.areHabitsDone) {
+      this.getClass = 'enrolled-past-day';
+      return this.getClass;
+    } else if (dayItem.hasHabitsInProgress && dayItem.numberOfDate < (dayItem.date.getDate() - 7) && !dayItem.areHabitsDone) {
+      this.getClass = 'unenrolled-past-day';
+      return this.getClass;
+    } else if (dayItem.hasHabitsInProgress && dayItem.areHabitsDone) {
+      this.getClass = 'enrolled-day';
+      return this.getClass;
+    } else if (dayItem.hasHabitsInProgress && !dayItem.areHabitsDone) {
+      this.getClass = 'unenrolled-day';
```

Comment:
use enum or const for class names

Reply:
also how about using switch instead of if's ?

Reply:
added enum.
switch not work in this case. I don't know why.